sqlalchemy-stubs/sql/functions.pyi

```diff
@@ -15,49 +16,60 @@ def register_function(identifier, fn, package: str = ...): ...
     packagenames: Any = ...
-    clause_expr: Any = ...
-    def __init__(self, *clauses, **kwargs) -> None: ...
+    clause_expr: Grouping[Any] = ...
+    def __init__(self, *clauses: Any, **kwargs: Any) -> None: ...
     @property
-    def columns(self): ...
+    def columns(self) -> ColumnCollection: ...
     @property
-    def clauses(self): ...
-    def over(self, partition_by: Optional[Any] = ..., order_by: Optional[Any] = ...,
-             rows: Optional[Any] = ..., range_: Optional[Any] = ...): ...
-    def within_group(self, *order_by): ...
-    def filter(self, *criterion): ...
-    def get_children(self, **kwargs): ...
-    def within_group_type(self, within_group): ...
-    def alias(self, name: Optional[Any] = ..., flat: bool = ...): ...
-    def select(self): ...
-    def scalar(self): ...
-    def execute(self): ...
-    def self_group(self, against: Optional[Any] = ...): ...
+    def clauses(self) -> ClauseList: ...
+    @overload
+    def over(self, partition_by: Optional[Union[str, ColumnElement[Any], Iterable[Union[str, ColumnElement[Any]]]]] = ...,
+             order_by: Optional[Union[str, ColumnElement[Any], Iterable[Union[str, ColumnElement[Any]]]]] = ...,
+             range_: Optional[Tuple[Optional[int], Optional[int]]] = ...) -> Over[_T]: ...
+    @overload
+    def over(self, partition_by: Optional[Union[str, ColumnElement[Any], Iterable[Union[str, ColumnElement[Any]]]]] = ...,
+             order_by: Optional[Union[str, ColumnElement[Any], Iterable[Union[str, ColumnElement[Any]]]]] = ...,
+             rows: Optional[Tuple[Optional[int], Optional[int]]] = ...) -> Over[_T]: ...
+    def within_group(self, *order_by: Union[str,
+                                            ColumnElement[Any],
+                                            Iterable[Union[str, ColumnElement[Any]]]]) -> WithinGroup[_T]: ...
+    @overload
+    def filter(self) -> FunctionElement[_T]: ...
+    @overload
+    def filter(self, criteria: Any, *criterion: Any) -> FunctionFilter[_T]: ...
+    def get_children(self, **kwargs: Any) -> Tuple[Grouping[Any]]: ...
+    def within_group_type(self, within_group: Any) -> Any: ...
+    def alias(self, name: Optional[str] = ..., flat: bool = ...) -> Alias: ...
+    def select(self) -> Select: ...  # type: ignore  # incompatible with FromClause.select
+    def scalar(self) -> Any: ...  # type: ignore  # incompatible with Executable.scalar
+    def execute(self) -> ResultProxy: ...  # type: ignore  # incompatible with Executable.scalar
+    def self_group(self, against: Optional[Any] = ...) -> Union[ColumnElement[_T], Grouping[_T]]: ...
 
 class _FunctionGenerator(object):
     opts: Any = ...
-    def __init__(self, **opts) -> None: ...
-    def __getattr__(self, name): ...
-    def __call__(self, *c, **kwargs): ...
+    def __init__(self, **opts: Any) -> None: ...
+    def __getattr__(self, name: str) -> Any: ...
+    def __call__(self, *c: Any, **kwargs: Any) -> Function[_T]: ...
```

Comment:
What is `_T` here? This class is not generic currently.

Reply:
This should probably be changed to `Any`